### PR TITLE
CI: Reduce Qt package compression level when not tagging

### PR DESCRIPTION
**Pull request by RytoEX**
### Description

The Windows Qt packaging step takes about 10 minutes on CI. We only need the maximum compression level on releases, so let's try using the default compression level on other events to see if that speeds up those CI runs.

### Motivation and Context

Want slightly faster Windows CI runs.

### How Has This Been Tested?

CI is the test. I want to compare the run time and artifact size.

### Types of changes

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [ ] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by RytoEX** [comment]
### Artifact Build Times

| Job                         | Before  | After   |
| --------------------------- | ------- | ------- |
| Package Windows Qt (5, x64) | 11m  7s |  4m 45s |
| Package Windows Qt (5, x86) |  9m 15s |  4m 22s |
| Package Windows Qt (6, x64) | 10m 41s |  3m 44s |
| Package Windows Qt (6, x86) | 10m 20s |  4m  6s |


### Artifact Sizes

| Artifact                    | Before  | After   |
| --------------------------- | ------- | ------- |
| qt5-windows-x64             | 149 MB  | 153 MB  |
| qt5-windows-x64-ReleasePDBs | 98.1 MB | 100 MB  |
| qt5-windows-x86             | 140 MB  | 145 MB  |
| qt5-windows-x86-ReleasePDBs | 96.5 MB | 98.8 MB |
| qt6-windows-x64             | 148 MB  | 151 MB  |
| qt6-windows-x64-ReleasePDBs | 103 MB  | 105 MB  |
| qt6-windows-x86             | 140 MB  | 143 MB  |
| qt6-windows-x86-ReleasePDBs | 101 MB  | 103 MB  |

References:
* https://github.com/obsproject/obs-deps/actions/runs/4346847616
* https://github.com/obsproject/obs-deps/actions/runs/4377664992

I'm starting to wonder if it's worth it at all to use the maximum compression level, even for tags, but perhaps we don't need to decide that now.

File: .github/actions/package-windows-qt/action.yml [hunk 1]
```diff
@@ -66,14 +66,16 @@ runs:
     - name: Create combined Qt archive
       shell: pwsh
       run: |
+        if ( '${{ github.event_name }}' -eq 'push' -and '${{ github.ref }}' -like '*refs/tags/*' ) { $CompressionLevel = '9' } else { $CompressionLevel = '5' }
```
Comment (PatTheMav) [inline .github/actions/package-windows-qt/action.yml:69]: This is fine, you can also check `github.ref_name` for the actual tag provided.

Comment (RytoEX) [reply .github/actions/package-windows-qt/action.yml:69]: I think I just copied this from our main.yml.

**Comment by PatTheMav** [comment]
> ### Artifact Build Times
> Job	Before	After
> Package Windows Qt (5, x64)	11m 7s	4m 45s
> Package Windows Qt (5, x86)	9m 15s	4m 22s
> Package Windows Qt (6, x64)	10m 41s	3m 44s
> Package Windows Qt (6, x86)	10m 20s	4m 6s
> ### Artifact Sizes
> Artifact	Before	After
> qt5-windows-x64	149 MB	153 MB
> qt5-windows-x64-ReleasePDBs	98.1 MB	100 MB
> qt5-windows-x86	140 MB	145 MB
> qt5-windows-x86-ReleasePDBs	96.5 MB	98.8 MB
> qt6-windows-x64	148 MB	151 MB
> qt6-windows-x64-ReleasePDBs	103 MB	105 MB
> qt6-windows-x86	140 MB	143 MB
> qt6-windows-x86-ReleasePDBs	101 MB	103 MB
> References:
> 
> * https://github.com/obsproject/obs-deps/actions/runs/4346847616
> * https://github.com/obsproject/obs-deps/actions/runs/4377664992
> 
> I'm starting to wonder if it's worth it at all to use the maximum compression level, even for tags, but perhaps we don't need to decide that now.

I'd argue the time savings are worth the few extra megabytes.

**Comment by RytoEX** [comment]
To clarify, this is fine to merge as-is, and we can adjust later if needed, per off-thread discussion.

**Pull request merged**